ipv8/messaging/anonymization/tunnel.py

```diff
@@ -186,6 +184,17 @@ def datagram_received(self, data, source):
         else:
             self.logger.warning("Dropping forbidden packets to exit socket with circuit_id %d", self.circuit_id)
 
+    def is_allowed(self, data):
+        is_bt = DataChecker.could_be_bt(data)
+        is_ipv8 = DataChecker.could_be_ipv8(data)
+
+        if not (is_bt and PEER_FLAG_EXIT_BT in self.overlay.settings.peer_flags) \
```

Comment:
Could you confirm this conditional is correct?

`is_bt = True` and `self.overlay.settings.peer_flags = [PEER_FLAG_EXIT_BT]`, leads to a return value of `False`.

Reply:
If `is_bt and PEER_FLAG_EXIT_BT in self.overlay.settings.peer_flags` is `True`, the if-statement will not be executed and `is_allowed` will return `True`. So that should be OK.

Having said that, this really needs a unit test. I'll add that.

Reply:
Ah right, I see. Thanks, must be suffering a heatstroke here 🥵